### DISCUSSION TITLE
Query re `Installing from sources`

### DISCUSSION THREAD

**Discussion by davidhay1969** [aquasecurity/kube-bench#856] [Development]
Hey

Reading [Installing from sources](https://github.com/aquasecurity/kube-bench#installing-from-sources), it's not 100% clear what the reader needs to do - I'm probably misunderstanding, but I was of the opinion that `go get` would "go get" the binary into $GOPATH/bin : -

```bash
ls -al $GOPATH/bin/
```

```text
total 16996
drwxr-x--- 2 root root     4096 Apr  8 09:30 .
drwxr-x--- 5 root root     4096 Apr  8 09:29 ..
-rwxr-x--- 1 root root 17392147 Apr  8 09:57 kube-bench
```

```bash
file $GOPATH/bin/kube-bench 
```

```text
/home/daveh/go/bin/kube-bench: ELF 64-bit MSB executable, IBM S/390, version 1 (SYSV), dynamically linked, interpreter /lib/ld64.so.1, not stripped
```

whereas, if we're really building from source, I'm guessing that we want to do something like this: -

```bash
mkdir -p $GOPATH/src/github.com/aquasecurity
cd $GOPATH/src/github.com/aquasecurity

git clone https://github.com/aquasecurity/kube-bench.git

cd $GOPATH/src/github.com/aquasecurity/kube-bench
go build -o kube-bench .
```

which gives us the binary, albeit in the cloned repo: -

```bash
file ./kube-bench 
```

```text
./kube-bench: ELF 64-bit MSB executable, IBM S/390, version 1 (SYSV), dynamically linked, interpreter /lib/ld64.so.1, not stripped
```

```bash
ls -al kube-bench 
```

```text
-rwxr-x--- 1 root root 17384496 Apr  8 10:01 kube-bench
```

So, to summarise, are we kinda combining the instructions to GET and BUILD the binary in the same section of the doc ?

Sorry if I'm misunderstanding, Dave

**Comment by yoavrotems** [comment]
Actually I think you are right and it is mixup, also there are some changes and go modules in 1.16 so maybe we should add coverage for that too

**Reply by davidhay1969** [reply]
More than happy to help update the doc if that'd help ?

**Reply by yoavrotems** [reply]
Sure you are very welcome ! There is also a new documentation site https://aquasecurity.github.io/kube-bench/dev

**Reply by davidhay1969** [reply]
Apologies for the *HUGE* delay @yoavrotems - have raised a PR #928 as a starting point .... happy to add more, if that'd help ?
